src/Lean/Elab/Extra.lean

```diff
@@ -535,7 +535,7 @@ def elabBinRelCore (noProp : Bool) (stx : Syntax) (expectedType? : Option Expr)
       if noProp then
         if (← withNewMCtxDepth <| isDefEq maxType (mkSort levelZero)) then
           maxType := Lean.mkConst ``Bool
-      let result ← toExprCore (← applyCoe tree maxType (isPred := true))
+      let result ← toExprCore (← applyCoe tree maxType (isPred := true)) none
```

Comment:
This is using `none` instead of `maxType` because there is a bug in `maxType` calculations -- they do not take into account annotations -- and this causes the `cleanup_forallTelescope` test to fail. I'll fix this in a followup PR.